w3w-autosuggest/admin/js/w3w-autosuggest-admin.js

```diff
@@ -0,0 +1,271 @@
+const COORD_REGEX = /(-?\d+(\.\d+)?)/
+
+function handleToggleTarget(target, checked) {
+  target.required = checked
+  target.disabled = !checked
+}
+
+function validateTarget(target, validCallback) {
+  target.classList.remove(['is-valid', 'is-invalid'])
+  if (!validCallback(target.value)) {
+    target.classList.add('is-invalid')
+    target.scrollTo()
+    return false
+  }
+
+  target.classList.add('is-valid')
+  return true
+}
+
+window.addEventListener('load', () => {
+
+  /**
+   * DROPDOWN MENU HANDLER
+   */
+  const dropdowns = document.querySelectorAll('.dropstart')
+  for (let i = 0; i < dropdowns.length; i++) {
+    const dropdown = dropdowns[i]
+    const dropdownToggle = dropdown.querySelector('.dropdown-toggle')
+    const dropdownMenu = dropdown.querySelector('.menu')
+    dropdownToggle.addEventListener('click', (e) => {
+      const target = e.target
+      if (target.classList.contains('expanded')) {
+        target.classList.remove('expanded')
+        dropdownMenu.classList.remove('d-block')
+        dropdownMenu.classList.add('d-none')
+      } else {
+        target.classList.add('expanded')
+        dropdownMenu.classList.remove('d-none')
+        dropdownMenu.classList.add('d-block')
+      }
+    })
+  }
+
+
+  /**
+   * API KEY HANDLER
+   */
+  const apiKeyInput = document.querySelector('input#w3w_api_key')
+  const apiKeyBtn = document.querySelector('button#api_key_btn')
+  function handleApiKeySetting(value) {
+    if (value.length !== 8) apiKeyBtn.disabled = true
+    else apiKeyBtn.disabled = false
```

Comment:
These two lines could be rewritten as `apiKeyBtn.disabled = value.length !== 8` but this is more of a stylistic choice

Reply:
True!